src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -177,12 +210,26 @@ public override bool TryGetPropertyValue(string name, out object value)
                 }
             }
 
-            PropertyAccessor<TStructuralType> cacheHit;
-            if (_allProperties.TryGetValue(name, out cacheHit))
+            if (this._deltaNestedResources.ContainsKey(name))
```

Comment:
>if (this._deltaNestedResources.ContainsKey(name) [](start = 12, length = 48)

the changes here make sense, So, again, i don't think we should have a new public API named "TryAddNestedResource", Please change TrySetPropertyValue to add the nested resource.